### ISSUE TITLE
🧑‍🌾 Flaky test: test_time_source.callback_handler_erasure

### ISSUE THREAD

**Issue by fujitatomoya** [ros2/rclcpp#2093]
## Bug report

**Required Info:**

- Operating System:
  - Ubuntu 22.04
- Installation type:
  - 
- Version or commit hash:
  - rolling:head
- DDS implementation:
  - Fast-DDS
- Client library (if applicable):
  - rclcpp

#### Steps to reproduce issue

```bash
root@tomoyafujita:~/ros2_ws/colcon_ws# colcon build --packages-up-to rclcpp
root@tomoyafujita:~/ros2_ws/colcon_ws# colcon test --event-handlers console_direct+ --retest-until-fail 5 --ctest-args -R test_time_source --packages-select rclcpp
```

#### Expected behavior

all tests are success.

#### Actual behavior

```bash
89: [ RUN      ] TestTimeSource.callback_handler_erasure
89: /root/ros2_ws/colcon_ws/src/ros2/rclcpp/rclcpp/test/rclcpp/test_time_source.cpp:499: Failure
89: Expected equality of these values:
89:   2
89:   cbo.last_precallback_id_
89:     Which is: 1
89: /root/ros2_ws/colcon_ws/src/ros2/rclcpp/rclcpp/test/rclcpp/test_time_source.cpp:500: Failure
89: Expected equality of these values:
89:   2
89:   cbo.last_postcallback_id_
89:     Which is: 1
89: [  FAILED  ] TestTimeSource.callback_handler_erasure (28 ms)
```

**Comment by fujitatomoya** [comment]
This flaky test is detected during https://github.com/ros2/rclcpp/pull/2092

**Comment by clalancette** [comment]
THis was fixed by #2111, so closing this.
